crates/types/src/lib.rs

```diff
@@ -924,6 +924,12 @@ entity_impl!(ModuleInternedTypeIndex);
 pub struct VMSharedTypeIndex(u32);
 entity_impl!(VMSharedTypeIndex);
 
+/// Index of a `call_indirect` instruction in a module, used for
+/// caching that callsite's target in the VMContext.
+#[derive(Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Debug, Serialize, Deserialize)]
+pub struct CallIndirectSiteIndex(u32);
```

Comment:
Oh I just noticed this new index type is in between the definition of VMSharedTypeIndex and the impl block defining methods for it. Can we move it before or after so that those two are together?